akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -126,6 +126,19 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
         errorCode should ===(ErrorCode.COMPRESSION_ERROR)
       }
 
+      "GOAWAY when the response has headers mid stream" in new SimpleRequestResponseRoundtripSetup {
+        val streamId = 0x1
+        emitRequest(streamId, HttpRequest(uri = "http://www.example.com/"))
+        expect[HeadersFrame]()
+
+        sendDATA(streamId, endStream = false, ByteString("wowdata"))
+        val headerBlock = encodeHeaders(Seq(RawHeader("X-Mid-Stream", "Wowsuchmid")))
+        sendFrame(HeadersFrame(streamId, endStream = false, endHeaders = true, headerBlock, None))
```

Comment:
Why would that be the second one? Isn't that interpreted as the first HEADERS of the response?

Reply:
Yeah, I don't know what I was thinking here TBH, this doesn't test the right thing at all.